agent/lm_agent/parsing/lmx.py

```diff
@@ -0,0 +1,117 @@
+"""
+Parser for LM-X
+"""
+import re
+
+HOSTWORD = r"[a-zA-Z0-9-]+"
+HOSTNAME = rf"{HOSTWORD}(\.{HOSTWORD})*"
+INT = r"\d+"
+VERSION = rf"{INT}\.{INT}"
+
+FEATURE_LINE = rf"^Feature: (?P<feature>\w+) Version: (?P<version>{VERSION}) Vendor: (?P<vendor>\S+)$"
+IN_USE_LINE = rf"^(?P<in_use>{INT}) of (?P<total>{INT}) license\(s\) used(:)?$"
+USAGE_LINE = (
+    rf"^(?P<in_use>{INT}) license\(s\) "
+    rf"used by (?P<user>\S+)@(?P<lead_host>{HOSTNAME}) "
+    rf"\[{INT}\.{INT}\.{INT}\.{INT}\]"
+)
+RX_FEATURE = re.compile(FEATURE_LINE)
+RX_IN_USE = re.compile(IN_USE_LINE)
+RX_USAGE = re.compile(USAGE_LINE)
+
+
+def parse_feature_line(line: str):
```

Comment:
I think typehints for the return types here would be good.

Reply:
Addressed in latest commit.